src/app/debug/table/table.component.html

```diff
@@ -98,42 +98,43 @@ <h4 class="progress-report-timout">[One or more reports are in progress for more
+            <ng-container [matColumnDef]="getMetadataNameFromHeader(header)">
+              <th mat-header-cell *matHeaderCellDef mat-sort-header
+                  data-cy-debug="metadataLabel"
+                  class="table-id-cell header-cell header-padding"
+                  [title]="header">
+                {{ getShortenedTableHeaderNames(header) }}
               </th>
-              @for (header of viewSettings.currentView.metadataLabels; track header) {
-                <th data-cy-debug="metadataLabel"
-                    class="table-id-cell header-cell header-padding"
-                    [mat-sort-header]="header"
-                    [title]="header">
-                  {{ getShortenedTableHeaderNames(header) }}
-                </th>
-              }
-            </tr>
-          </thead>
-          <tbody data-cy-debug="tableBody">
-            @for (row of sortedReportMetadata.slice(0, tableSettings.displayAmount); let index = $index; track row.storageId) {
-              <tr class="table-row"
-                  [ngClass]="{'highlight': row.storageId === selectedReportStorageId}"
-                  [style.background-color]="getStatusColor(row)"
-                  [attr.data-cy-record-table-index]="index"
-                  (click)="openSelectedReport(row.storageId);"
-              >
-                <td class="table-row-checkbox" [style.padding]="getTableSpacing()">
-                  <input type="checkbox" [checked]="row.checked" [style.width]="getCheckBoxSize()" [style.height]="getCheckBoxSize()" (click)="toggleCheck(row)">
-                </td>
-                @for (metadataName of viewSettings.currentView.metadataNames; track metadataName) {
-                  <td
-                    class="table-cell"
-                    [title]="metadataName">
-                    {{ getMetadata(row, metadataName)|tableCellShortener }}</td>
-                }
-              </tr>
-            }
-          </tbody>
+              <td mat-cell *matCellDef="let row"
+                class="table-cell"
+                [style.padding]="getTableSpacing()"
+                [title]="header">
+                {{ getMetadata(row, getMetadataNameFromHeader(header))|tableCellShortener }}</td>
+            </ng-container>
+          }
+
+          <!--Values are passed to their respective header and row elements-->
+          <tr mat-header-row *matHeaderRowDef="getDisplayedColumnNames(viewSettings.currentView.metadataLabels)"></tr>
+          <tr mat-row *matRowDef="let row; let rowIndex = index; columns: getDisplayedColumnNames(viewSettings.currentView.metadataLabels)"
+              class="table-row"
+              [ngClass]="{'highlight': row.storageId === selectedReportStorageId}"
+              [attr.data-cy-record-table-index]="rowIndex"
+              [style.background-color]="getStatusColor(row)"
+              (click)="openSelectedReport(row.storageId);"
+          ></tr>
```

Comment:
I don't understand why you've split the table into three parts like that?

Reply:
I followed the example code from the angular docs on the Material Table. The only big thing that I changed from the example code is that this creates headers dynamically depending on the labels in the viewSettings.currentView.metadataLabels